### PR TITLE
Fix a Couple bugs related to pivot exports

### LINKED ISSUE #48158: xlsx Export doesn't respect custom column formating, instead format is carried over to the next column
### Describe the bug

Below, attached is a SS, which shows the question result preview, along with the column format (which is set to Percentage)
image

Now when I download the question into xlsx and open it in excel, then the formating shown in the excel file is very weird. 
image

Interesting, if I download the question into CSV format and import it in excel, then the formating is shown correctly. 
CSV preview view in MAC -> 
image

After Importing the CSV into Excel in MAC ->
image

Seems like its working fine when I Download the question in CSV format, but something is broken when I Download the question in xlsx format

### To Reproduce

Metabase version ->  Metabase 0.50.27

1. Open an already created Question, which has Custom Column in it, and Custom Column setting is set to Percentage
2. Now download the Question in xlsx format, and open it in Excel.
3. One can see that the Custom Column which was set to Percentage is not having Percentage format, but the column next to it is having the Percentage formating.

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
- Browser - Version 129.0.6668.71 (Official Build) (arm64)
- OS - macOS Sonoma 14.6.1
- Databases - Postgres
- Metabase Version 0.50.27
```

### Severity

Annoying, as our Clients are not able to download the data in xlsx format. And now we have to guide them on how to use the CSV format and get the data. 

### Additional context

_No response_

### LINKED ISSUE #48849: Downloading Pivot Table doesn't show data only column headers - Oracle Specific
### Describe the bug

When you download a Pivot (any pivot config) there are no rows visible in the csv/xlsx/json files ... This is what you get:

Image

Only happens for Oracle based connections

### To Reproduce

Connect to an Oracle DB - Luis has a handy lab for Linux https://github.com/paoliniluis/metabase-oracle

1. Go to Orders -> Create a couple of aggregates:

Image

2. You will get something like this:

Image

3. Then download the pivot and notice the empty file:

Image

### Expected behavior

The file is populated with data

### Logs

Nothing that is relevant

### Information about your Metabase installation

50.28

### Severity

Breaks download functionality for pivot tables which is client facing. 

### Additional context

_No response_

### LINKED ISSUE #48868: Latest version, download json file to convert numeric format to string
### Describe the bug

Latest version, download json file to convert numeric format to string

version:0.50.x    --- bug
```
[
{"QTY":"1"}
]
```

version:0.47.9   --- ok
```
[
{"QTY":1}
]
```

### To Reproduce

Image

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

centos 
java -jar metabase.jar

### Severity

bug

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by adam-james-v**
- [x] pivot-grouping value, when not an int (in the case of Oracle dbs), would cause exports to contain zero rows. This fixes that fixes #48849
 - [x] json formatted/unformatted for the dataset API endpoints (unsaved questions) was not being used, so all exports were formatted fixes #48868
 - [x] pivot-grouping column removal should also remove cell styles in xlsx to apply the proper styles to cols after the grouping column fixes #48158

### CODE REVIEW

File: src/metabase/api/dataset.clj [hunk 1]
```diff
@@ -144,7 +146,9 @@
                                           (dissoc :constraints)
                                           (update :middleware #(-> %
                                                                    (dissoc :add-default-userland-constraints? :js-int-to-string?)
-                                                                   (assoc :process-viz-settings? true
+                                                                   (assoc :format-rows?           (or format_rows false)
+                                                                          :pivot?                 (or pivot_results false)
+                                                                          :process-viz-settings? true
```
Comment (escherize) [inline src/metabase/api/dataset.clj:151]: nit: indentation 
```suggestion
                                                                          :process-viz-settings?  true
```

File: src/metabase/query_processor/streaming/xlsx.clj [hunk 2]
```diff
@@ -678,7 +680,7 @@
                                    pivot-grouping-key (m/remove-nth pivot-grouping-key))
               {:keys [sheet]}    @workbook-data]
           (when (or (not group)
-                    (= group 0))
+                    (= qp.pivot.postprocess/NON_PIVOT_ROW_GROUP (int group)))
```
Comment (escherize) [inline src/metabase/query_processor/streaming/xlsx.clj:683]: what does the call to `int` do here? Can it be a double? Is it worth checking for `number?` in the `when` on the line above?

**Review by escherize — APPROVED** [review]
Looks good. Some small things but nothing blocking

File: src/metabase/api/dataset.clj (hunk 1 above)
Comment (escherize) [reply src/metabase/api/dataset.clj:151]: Actually it should be like:

```
                                                                   (assoc :format-rows?          (or format_rows false)
                                                                          :pivot?                (or pivot_results false)
                                                                          :process-viz-settings? true                                                           
```

File: src/metabase/query_processor/streaming/xlsx.clj (hunk 2 above)
Comment (adam-james-v) [reply src/metabase/query_processor/streaming/xlsx.clj:683]: In most cases, the value will be an int. But, there was a bug with Oracle dbs where the value was a BigDecimal. 
So, calling int just coerces things into int for simple comparison. The order of checking (not group) first should always guarantee that we don't try `(int nil)`, and the pivot-grouping value is actually set by some code in the pivot query processor code, so we know it'll never be a string. It'll always be a number or nil.

File: src/metabase/api/dataset.clj (hunk 1 above)
Comment (adam-james-v) [inline src/metabase/api/dataset.clj:151]:
```suggestion
                                                                   (assoc :format-rows?          (or format_rows false)
                                                                          :pivot?                (or pivot_results false)
                                                                          :process-viz-settings? true
```

**Pull request merged**